src/ICS20Transfer.sol

```diff
@@ -168,4 +153,124 @@ contract ICS20Transfer is IIBCApp, IICS20Transfer, IICS20Errors, Ownable, Reentr
             revert ICS20UnexpectedERC20Balance(expectedEndingBalance, actualEndingBalance);
         }
     }
+
+    function _unwrapSendPacketData(IICS26RouterMsgs.Packet calldata packet)
+        private
+        view
+        returns (ICS20Lib.SendPacketData memory)
+    {
+        ICS20Lib.PacketDataJSON memory packetData = ICS20Lib.unmarshalJSON(packet.data);
+        ICS20Lib.SendPacketData memory receivePacketData = ICS20Lib.SendPacketData({
```

Comment:
I just wanted to mention why this is being used instead of local variables further down. I implemented this first with a lot of local variables to make things as readable and nice as I could, but encountered some "stack too deep" issue because there were too many local variables. So I ended up using the return value as a way to avoid that issue.

There are a couple of alternatives to this that can be considered:
* Splitting up the function more
* Using brackets to scope temporary variables along the way so they don't affect the stack too much (can be a bit ugly, but might be more readable than the current way).

Might be some other way to optimise this, but those were the options I could see, at least.

Reply:
Yeah, I understand what you mean. The main reason I ended up with doing that is because of the difference in addresses/string between sender and receiver and figuring out source/not source being kind of opposite. The other option is to only use the ICS20Lib.PacketDataJSON, or add a single slightly expanded version that keeps all fields the same (but would have to extract sender/receiver addresses when they are needed.

Reply:
Why can't these addresses be just returned as additional return arguments instead of whole structs?

Reply:
Consider guys that I if we have the stack too deep problem, if we really need the variables the only real solution we have is to split the logic over multiple contract/libraries. 
- Overall I agree we should always try to keep things as minimal as we can IF we can. 
- We should use nat spec (just give the function to gpt-4 or whatever AI you use and ask for it, don't spend time doing it manually)
- We should create a specific library to handle this unwrapping things

Reply:
That is also an option. Will look into it and see what makes most sense